pmd-java/src/main/java/net/sourceforge/pmd/lang/java/oom/Metrics.java

```diff
@@ -0,0 +1,88 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.java.oom;
+
+import static net.sourceforge.pmd.lang.java.oom.visitor.MetricsVisitorFacade.TOP_LEVEL_PACKAGE;
+
+import net.sourceforge.pmd.lang.java.ast.ASTClassOrInterfaceDeclaration;
+import net.sourceforge.pmd.lang.java.ast.ASTMethodOrConstructorDeclaration;
+import net.sourceforge.pmd.lang.java.ast.QualifiableNode.QualifiedName;
+import net.sourceforge.pmd.lang.java.oom.metrics.AtfdMetric;
+import net.sourceforge.pmd.lang.java.oom.metrics.WmcMetric;
+
+
+/**
+ * User bound façade of the Metrics Framework. Provides a uniform interface for the calculation of
+ * metrics.
+ *
+ * @author Clément Fournier
+ */
+public class Metrics {
+
+    private Metrics() { // Cannot be instantiated
+
+    }
+
+    /**
+     * Computes a metric identified by its code on the class AST node being
+     * passed.
+     */
+    public static double get(ClassMetricKey key, ASTClassOrInterfaceDeclaration node) {
+        QualifiedName qname = node.getQualifiedName();
+        double memoized = TOP_LEVEL_PACKAGE.getMemo(key, qname); // TODO
+
+        return memoized == Double.NaN ? key.getCalculator().computeFor(node, TOP_LEVEL_PACKAGE) : memoized;
```

Comment:
I only thought about a possibility, where the calculated value of a metric would be already NaN - then we would not be able to cache it and would always (re)calculate it. But maybe that's an edge case. Anyway, I believe we could change this later in any way without changing the API of Metrics - so it's a pure implementation detail. We could even say, we add a type "MemoizedMetric" which has "hasValue" - hm.... not sure whether this now makes sense :) Well, I'd suggest to revisit this once you're tackling the actual implementation in PackageStats.